WordPress/Classes/Networking/AccountSettingsRemote.swift

```diff
@@ -1,12 +1,78 @@
+import AFNetworking
 import Foundation
+import RxSwift
 
 class AccountSettingsRemote: ServiceRemoteREST {
-    func getSettings(success success: AccountSettings -> Void, failure: ErrorType -> Void) {
+    static let remotes = NSMapTable(keyOptions: .StrongMemory, valueOptions: .WeakMemory)
+
+    /// Returns an AccountSettingsRemote with the given api, reusing a previous
+    /// remote if it exists.
+    static func remoteWithApi(api: WordPressComApi) -> AccountSettingsRemote {
+        // We're hashing on the authToken because we don't want duplicate api
+        // objects for the same account.
+        //
+        // In theory this would be taken care of by the fact that the api comes
+        // from a WPAccount, and since WPAccount is a managed object Core Data
+        // guarantees there's only one of it.
+        // 
+        // However it might be possible that the account gets deallocated and
+        // when it's fetched again it would create a different api object.
+        let key = api.authToken.hashValue
+        // FIXME: not thread safe
+        // @koke 2016-01-21
```

Comment:
I think this is one of those cases why services have been kept ephemeral to prevent issues like this.

I'd rather use the uniqueness of WordPressComApi which should come up with its own hashValue that is based upon a set of business logic conditions. This makes AccountSettingsRemote know too much about the api class I think. Then, in theory, two instances of WordPressComApi that are logically equivalent (same auth token, same username, etc) are interchangeable.